external/mmdetection/detection_tasks/apis/detection/inference_task.py

```diff
@@ -218,15 +218,17 @@ def _add_predictions_to_dataset(self, prediction_results, dataset, confidence_th
                             polygon = Polygon(points=points)
                             if polygon.get_area() > 1e-12:
                                 shapes.append(Annotation(polygon, labels=labels, id=ID(f"{label_idx:08}")))
+                    if add_saliency_map and features is not None:
+                        features.append(draw_instance_segm_saliency_map(shapes, dataset_item, self._labels))
             else:
                 raise RuntimeError(
                     f"Detection results assignment not implemented for task: {self._task_type}")
 
             dataset_item.append_annotations(shapes)
-
-            if feature_vector is not None:
-                active_score = TensorEntity(name="representation_vector", numpy=feature_vector)
-                dataset_item.append_metadata_item(active_score, model=self._task_environment.model)
+            if features is not None:
```

Comment:
> The assumption is that "feature_vector" is always present in the outputs, right? Is it the same for PyTorch task?


>
> Yes, it's a representation of backbone+neck output. For Pytorch the same

How could `features` be None then?

Reply:
If `dump_features=False`, we append to the `feature_list` None [here](https://github.com/openvinotoolkit/training_extensions/blob/7957b88d4e73090b129f511694736f21e26c5b16/external/mmdetection/detection_tasks/apis/detection/inference_task.py#L297), so that's how `features` can be `None`

Reply:
This is can never happen in current implementation.

Reply:
It can happened during training, then we do evaluation ([here](https://github.com/openvinotoolkit/training_extensions/blob/f66fa45d5842d65ea4a0b4c1b77f6c170befa3b1/external/mmdetection/detection_tasks/apis/detection/train_task.py#L156))

Reply:
I see, my bad. Thanks.